### ISSUE TITLE
Paste text into a text file crashes JVM from libcairo.so

### ISSUE THREAD

**Issue by edmiikk** [eclipse-platform/eclipse.platform.swt#547]
Eclipse IDE for Java Developers (includes Incubating components)

Version: 2022-12 (4.26.0)
Build id: 20221201-1913

Paste text into a text editor panel or a wikitext editor panel can result in a fatal crash. Text from a Chrome window. e.g. a URL (twice) or copy to clipboard code example (once).

<code>
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fc75c9453c4, pid=203021, tid=203022
#
# JRE version: OpenJDK Runtime Environment Temurin-17.0.6+10 (17.0.6+10) (build 17.0.6+10)
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (17.0.6+10, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libcairo.so.2+0x643c4]  cairo_region_num_rectangles+0x4
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/fredagarb/tools/eclipse_2022/eclipse/hs_err_pid203021.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
</code>

**Comment by SyntevoAlex** [comment]
Judging from `cairo_region_num_rectangles+0x4` this is probably a duplicate of https://github.com/eclipse-platform/eclipse.platform.swt/issues/158

**Comment by edmiikk** [comment]
One for each crash. 
[hs_err_pid186451.log.gz](https://github.com/eclipse-platform/eclipse.platform.swt/files/10676253/hs_err_pid186451.log.gz)
[hs_err_pid197883.log.gz](https://github.com/eclipse-platform/eclipse.platform.swt/files/10676259/hs_err_pid197883.log.gz)
[hs_err_pid203021.log.gz](https://github.com/eclipse-platform/eclipse.platform.swt/files/10676265/hs_err_pid203021.log.gz)

**Comment by edmiikk** [comment]
Development setup
Ubuntu version is 22.04.1 LTS, 
Wayland windowing system.

**Comment by akurtakov** [comment]
Is this still reproducible with Eclipse 2026-06?

**Comment by peutch** [comment]
Hi,
For sure, not anymore on my side!
